test/forms/register/establishers/individual/EstablisherNinoFormProviderSpec.scala

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.register.establishers.individual
+
+import forms.FormSpec
+import models.EstablisherNino
+import org.apache.commons.lang3.RandomStringUtils
+
+class EstablisherNinoFormProviderSpec extends FormSpec {
+
+  val requiredKey = "messages__error__has_nino_establisher"
+  val requiredNinoKey = "messages__error__nino"
+  val requiredReasonKey = "messages__establisher__no_nino"
+  val invalidNinoKey = "messages__error__nino_invalid"
+
+  val reasonMaxLength = 150
+  val reasonInvalidLength = 151
+  val formProvider = new EstablisherNinoFormProvider()()
+  val validData:Map[String,String] = Map(
+    "establisherNino.hasNino" ->"true",
+    "establisherNino.nino" -> "AB020202A"
+  )
+
+  "EstablisherNino form provider" must {
```

Comment:
We should either be testing the mapping, or the logic that we're testing here should be in this class.